lib/matplotlib/axis.py

```diff
@@ -57,26 +58,12 @@ class Tick(martist.Artist):
     """
     def __init__(
         self, axes, loc, *,
-        size=None,  # points
-        width=None,
-        color=None,
-        tickdir=None,
-        pad=None,
-        labelsize=None,
-        labelcolor=None,
-        labelfontfamily=None,
-        zorder=None,
-        gridOn=None,  # defaults to axes.grid depending on axes.grid.which
-        tick1On=True,
-        tick2On=True,
-        label1On=True,
-        label2On=False,
-        major=True,
-        labelrotation=0,
-        grid_color=None,
-        grid_linestyle=None,
-        grid_linewidth=None,
-        grid_alpha=None,
+        size=None, width=None, color=None, major=True, pad=None, zorder=None,  # global
```

Comment:
The size in points must stay.

I'd have a slight preference to keep the one-parameter-per-line format and reorder in there:
```
# global
size=None,  # points
width=None,
...
# tick props
tickdir=None,
...
```

But that's personal taste and not a blocker.

Reply:
I agree about the size in points comment staying.